soroban-env-host/src/host/conversion.rs

```diff
@@ -150,4 +155,87 @@ impl Host {
         }
         self.storage_key_from_rawval(k)
     }
+
+    // TODO: impl a `TryFrom` trait once the "metered_" class is ready
+    pub(crate) fn scobj_from_bigint(&self, bi: &BigInt) -> Result<ScObject, HostError> {
+        let (sign, data) = bi.to_bytes_be();
+        match sign {
+            Sign::Minus => Ok(ScObject::BigInt(ScBigInt::Negative(
+                self.map_err(data.try_into())?,
+            ))),
+            Sign::NoSign => Ok(ScObject::BigInt(ScBigInt::Zero)),
+            Sign::Plus => Ok(ScObject::BigInt(ScBigInt::Positive(
+                self.map_err(data.try_into())?,
+            ))),
+        }
+    }
+
+    fn event_topic_from_rawval(&self, topic: RawVal) -> Result<ScVal, HostError> {
+        self.charge_budget(CostType::ValXdrConv, 1)?;
+        if topic.is_u63() {
+            Ok(ScVal::U63(unsafe { topic.unchecked_as_u63() }))
+        } else {
+            match topic.get_tag() {
+                Tag::Object => {
+                    unsafe {
+                        self.unchecked_visit_val_obj(topic, |ob| {
+                            // charge budget
+                            let sco = match ob {
+                                None => Err(self.err_status(ScHostObjErrorCode::UnknownReference)),
+                                Some(ho) => match ho {
+                                    // TODO: use more event-specific error codes than `UnexpectedType`
+                                    HostObject::Vec(_) => {
+                                        Err(self.err_status(ScHostObjErrorCode::UnexpectedType))
+                                    }
+                                    HostObject::Map(_) => {
+                                        Err(self.err_status(ScHostObjErrorCode::UnexpectedType))
+                                    }
+                                    HostObject::Bin(b) => {
+                                        if b.len() > 32 {
```

Comment:
💡 I think it might be worth giving a name to this 32, by moving it into a constant. It's a rather magic number.

Reply:
Fixed. Thank you.